applicationLibrary/molecularDynamics/tests/singleSiteTests/ATMFunctorTestNoGlobals.cpp

```diff
@@ -223,325 +237,200 @@ TYPED_TEST_P(ATMFunctorTestNoGlobals, testSoANoGlobalsATM) {
-      EXPECT_DOUBLE_EQ(f2[1], 0);
-      EXPECT_DOUBLE_EQ(f2[2], 0);
+    if (newton3 or soaFunctorType == TestType::SoAFunctorType::single or
+        soaFunctorType == TestType::SoAFunctorType::pair21 or soaFunctorType == TestType::SoAFunctorType::verlet) {
+      // If particle 2 is in the same cell as particle 1, its forces should already be calculated
+      const auto expectedForceIter1P2 = mixing ? this->expectedForceMixingP2 : this->expectedForceNonMixingP2;
+      testNonZeroForce(*f2, expectedForceIter1P2, ATMFunctorTest::to_string(soaFunctorType));
+    } else {  // Particle 2 is in a different cell, so the forces are zero when newton3 == false
+      testZeroForce(*f2, ATMFunctorTest::to_string(soaFunctorType));
     }
 
     // force of particle 3
-    std::array<double, 3> f3 = {0., 0., 0.};
-    switch (interactionType) {
-      case TestType::InteractionType::verlet:
-      case TestType::InteractionType::own: {
-        auto iter = ++cell1.begin();
-        f3 = (++iter)->getF();
-        break;
-      }
-      case TestType::InteractionType::pair21:
-        f3 = cell2.begin()->getF();
-        break;
-      case TestType::InteractionType::pair12:
-        f3 = (++cell2.begin())->getF();
-        break;
-      case TestType::InteractionType::triple:
-        f3 = cell3.begin()->getF();
-        break;
+    auto *f3 = &(++(++cell1.begin()))->getF();
+    if (soaFunctorType == TestType::SoAFunctorType::pair21) {
+      f3 = &cell2.begin()->getF();
+    } else if (soaFunctorType == TestType::SoAFunctorType::pair12) {
+      f3 = &(++cell2.begin())->getF();
+    } else if (soaFunctorType == TestType::SoAFunctorType::triple) {
+      f3 = &cell3.begin()->getF();
     }
 
-    // if the interactionType is own, then the forces of the third particle should always be calculated!
-    if (newton3 or interactionType == TestType::InteractionType::own or
-        interactionType == TestType::InteractionType::verlet) {
-      if (mixing) {
-        EXPECT_NEAR(f3[0], this->expectedForceMixingP3[0], this->absDelta);
-        EXPECT_NEAR(f3[1], this->expectedForceMixingP3[1], this->absDelta);
-        EXPECT_NEAR(f3[2], this->expectedForceMixingP3[2], this->absDelta);
-      } else {
-        EXPECT_NEAR(f3[0], this->expectedForceP3[0], this->absDelta);
-        EXPECT_NEAR(f3[1], this->expectedForceP3[1], this->absDelta);
-        EXPECT_NEAR(f3[2], this->expectedForceP3[2], this->absDelta);
-      }
+    if (newton3 or soaFunctorType == TestType::SoAFunctorType::single or
+        soaFunctorType == TestType::SoAFunctorType::verlet) {
+      const auto expectedForceIter1P3 = mixing ? this->expectedForceMixingP3 : this->expectedForceNonMixingP3;
+      testNonZeroForce(*f3, expectedForceIter1P3, ATMFunctorTest::to_string(soaFunctorType));
     } else {
-      EXPECT_DOUBLE_EQ(f3[0], 0);
-      EXPECT_DOUBLE_EQ(f3[1], 0);
-      EXPECT_DOUBLE_EQ(f3[2], 0);
+      testZeroForce(*f3, ATMFunctorTest::to_string(soaFunctorType));
     }
```

Comment:
I think this way of looping through the three particles is quite messy. Maybe, you could simply create an array/vector of particles, and simply loop through each cell, adding particles to the array, or skipping the cell if empty. Then apply the tests. Alternatively, simply write the whole thing with a type-case switch - I feel the advantages gained from reduced code duplication you have here are lost from having all these complex `if`-statements and branching. 

In any case, also remove the pointers - you can just use references.